packages/integration-karma/test/light-dom/slotting/index.spec.js

```diff
@@ -1,55 +1,104 @@
+            nodes['5'],
+            nodes['4'],
+            nodes['3'],
+            nodes['2'],
+            nodes['1'],
+            nodes['container-default-slot-bottom'],
+            nodes['container-lower-slot-top'],
+            nodes['container-lower-slot-default'],
+            nodes['container-lower-slot-bottom'],
+        ]);
     });
-});
 
-describe('imperative slotted content', () => {
-    beforeEach(() => {
-        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', true);
+    it('shadow container, light consumer', () => {
+        const nodes = createTestElement('x-light-consumer', LightConsumer);
+
+        const expected = process.env.NATIVE_SHADOW // native shadow doesn't output slots in innerHTML
+            ? '<x-shadow-container><p data-id="light-consumer-text">Hello from Light DOM</p></x-shadow-container>'
+            : '<x-shadow-container><slot><p data-id="light-consumer-text">Hello from Light DOM</p></slot></x-shadow-container>';
+        expect(nodes['x-light-consumer'].innerHTML).toEqual(expected);
     });
 
-    afterEach(() => {
-        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', false);
+    it('light container, shadow consumer', () => {
+        const nodes = createTestElement('x-shadow-consumer', ShadowConsumer);
+
+        expect(nodes['x-shadow-consumer'].shadowRoot.innerHTML).toEqual(
+            '<x-light-container><p data-id="container-upper-slot-top">Upper slot top</p><p data-id="container-upper-slot-default">Upper slot default</p><p data-id="container-upper-slot-bottom">Default slot bottom</p><p data-id="container-default-slot-top">Default slot top</p><p data-id="shadow-consumer-text">Hello from Shadow DOM</p><p data-id="container-default-slot-bottom">Default slot bottom</p><p data-id="container-lower-slot-top">Lower slot top</p><p data-id="container-lower-slot-default">Lower slot default</p><p data-id="container-lower-slot-bottom">Lower slot bottom</p></x-light-container>'
+        );
     });
 
-    [
-        ['Node.prototype.insertBefore', (elm) => elm.insertBefore(document.createElement('div'))],
-        ['Node.prototype.appendChild', (elm) => elm.appendChild(document.createElement('div'))],
-        ['Node.prototype.removeChild', (elm) => elm.removeChild(elm.firstChild)],
-        [
-            'Node.prototype.replaceChild',
-            (elm) => elm.replaceChild(elm.firstChild, document.createElement('div')),
-        ],
-    ].forEach(([operation, fn]) => {
-        it(`throws an error when invoking ${operation}`, () => {
-            const element = createElement('x-container', { is: Container });
-            document.body.appendChild(element);
-
-            expect(() => {
-                fn(element);
-            }).toThrowError(
-                /Invalid DOM operation on <x-container>\. Light DOM component don't allow imperative slotted content\./
-            );
-        });
+    it('removes slots properly', async () => {
```

Comment:
I would recommend also adding a test similar to this one where a light DOM consumer pass some slotted content and where the slot is conditionally added and remove:

```html
<template lwc:render-mode="light">
  <x-conditional-slot>
    <div>I might be rendered or not.</div>
  </x-conditional-slot>
</template>
```

This test doesn't capture the fact that slotted content might be passed.